curriculum/challenges/english/25-front-end-development/quiz-javascript-problem-solving-and-algorithmic-thinking/66edd4bc0ce399f475ad6f9b.md

```diff
@@ -17,439 +17,439 @@ Answer all of the questions below correctly to pass the quiz.
+Use recursion to reverse digits.
 
 #### --answer--
 
-Placeholder answer
+Use a loop to extract digits from the integer and reconstruct the reversed integer.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Write a function that generates Pascal's Triangle up to `n` rows. For example, for `n = 3`, return `[[1], [1, 1], [1, 2, 1]]`.
 
 #### --distractors--
 
-Placeholder distractor 1
+Generate rows by using a function that outputs prime numbers and maps them to each row.
 
 ---
 
-Placeholder distractor 2
+Use a loop to generate each row by multipling adjacent elements from the previous row.
 
 ---
 
-Placeholder distractor 3
+Use `Math.random()` to generate the rows.
 
 #### --answer--
 
-Placeholder answer
+Use a loop to generate each row by adding adjacent elements from the previous row.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which algorithm is most efficient for sorting an array of size n containing numbers from `1` to `n`?
 
 #### --distractors--
 
-Placeholder distractor 1
+Bubble Sort
 
 ---
 
-Placeholder distractor 2
+Merge Sort
 
 ---
 
-Placeholder distractor 3
+Quick Sort
 
 #### --answer--
 
-Placeholder answer
+Cyclic Sort
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which algorithmic approach is best for solving the problem of finding the minimum number of coins for a given amount when only one type of coin is provided?
 
 #### --distractors--
 
-Placeholder distractor 1
+Dynamic Programming
 
 ---
 
-Placeholder distractor 2
+Divide and Conquer
 
 ---
 
-Placeholder distractor 3
+Backtracking
 
 #### --answer--
 
-Placeholder answer
+Greedy Algorithm
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Which of the following problems can be solved efficiently using the two-pointer technique?
 
 #### --distractors--
 
-Placeholder distractor 1
+Sorting an array.
 
 ---
 
-Placeholder distractor 2
+Traversing a tree level by level.
 
 ---
 
-Placeholder distractor 3
+Finding the middle element of a linked list.
 
 #### --answer--
 
-Placeholder answer
+Finding two numbers in a sorted array that sum up to a target value.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+Given a Roman numeral string, convert it to its corresponding integer value. For example, `"IX"` should return `9`.
 
 #### --distractors--
 
-Placeholder distractor 1
+Parse the string from left to right and add the values of individual numerals.
 
 ---
 
-Placeholder distractor 2
+Parse each character and multiply by its index.
 
 ---
 
-Placeholder distractor 3
+Use the `parseInt()` function on the Roman numeral string.
 
 #### --answer--
 
-Placeholder answer
+Parse the string from right to left, adding the values of individual numerals and subtracting when a smaller numeral comes before a larger numeral.
 
 ### --question--
 
 #### --text--
 
-Placeholder question
+How can one check if a given string is a palindrome (the same forwards and backwards), for example `"madam"`?
```

Comment:
this is also one of the projects in the curriculum

Reply:
@naomi-lgbt what to do? should I replace this?

Reply:
This should be fine to keep. This quiz comes long after that project.